landside/catkin_ws/src/gui/src/gui/camera_status_widget.py

```diff
@@ -0,0 +1,417 @@
+        }
+
+        self.checking = {}
+        for camera_type in CameraStatusDataType:
+            self.checking[camera_type] = False
+
+        self.status_logs = {}
+        for camera_type in CameraStatusDataType:
+            self.status_logs[camera_type] = []
+
+        self.status_table.horizontalHeader().setSectionResizeMode(QHeaderView.Stretch)
+
+        self.timer = QTimer(self)
+        self.timer.timeout.connect(self.timer_check)
+        self.timer.start(100)
+
+        self.subscriber = rospy.Subscriber(
+            CAMERA_STATUS_DATA_TYPE_INFORMATION[CameraStatusDataType.PING]["topic_name"],
+            DiagnosticArray,
+            self.ping_response
+        )
+
+        self.init_table()
+
+        rospy.loginfo('Camera Status Widget successfully initialized')
+
+    @pyqtProperty(str)
+    def hostname(self):
+        return self.ping_hostname
+
+    @hostname.setter
+    def hostname(self, value):
+        self.ping_hostname = value
+
+    @pyqtProperty(int)
+    def channel(self):
+        return self.usb_channel
+
+    @channel.setter
+    def channel(self, value):
+        self.usb_channel = value
+
+    def timer_check(self):
+        self.check_buttons_enabled()
+        self.check_ping_publisher()
+
+    def check_ping_publisher(self):
+        master = rosgraph.Master('/rostopic')
+        pubs, _ = rostopic.get_topic_list(master=master)
+        for topic_name, _, publishing_nodes in pubs:
+            if topic_name == CAMERA_STATUS_DATA_TYPE_INFORMATION[CameraStatusDataType.PING]["topic_name"] and \
+                    len(publishing_nodes) > 0:
+                if self.subscriber is None:
+                    self.create_new_subscriber()
+                return
+
+        if self.subscriber is not None:
+            self.remove_subscriber()
+
+    def create_new_subscriber(self):
+        self.subscriber = rospy.Subscriber(
+            CAMERA_STATUS_DATA_TYPE_INFORMATION[CameraStatusDataType.PING]["topic_name"],
+            DiagnosticArray,
+            self.ping_response
+        )
+
+    def remove_subscriber(self):
+        self.subscriber.unregister()
+        self.subscriber = None
+
+    def check_buttons_enabled(self):
+        service_list = rosservice.get_service_list()
+        for camera_type in self.check_camera_buttons:
+            self.check_camera_buttons[camera_type].setEnabled(
+                CAMERA_STATUS_DATA_TYPE_INFORMATION[camera_type]['service_name'] in service_list and
+                not self.checking[camera_type]
+            )
+
+    def open_conection_log(self):
+        self.log = CameraStatusLog(self.data_updated, self.status_logs)
+        self.log.exec()
+
+    def check_camera_connection(self, camera_type):
+        if camera_type == CameraStatusDataType.MONO:
+            call_connect_camera_service = CallConnectCameraService(camera_type, self.channel)
+        else:
+            call_connect_camera_service = CallConnectCameraService(camera_type)
```

Comment:
Is there a way to generalize this? Everything else seems very camera agnostic, so this hard check for MONO kind of sticks out

Reply:
Looks okay to me